app/views/manuals/confirm_publish.html.erb

```diff
@@ -29,10 +29,12 @@
 
 <div class="govuk-button-group govuk-!-margin-bottom-6">
   <%= form_tag(publish_manual_path(manual), method: :post) do %>
-    <%= render "govuk_publishing_components/components/button", {
-      text: "Publish",
-      name: "submit",
-    } %>
+    <% if allow_publish?(manual, slug_unique) %>
+      <%= render "govuk_publishing_components/components/button", {
+        text: "Publish",
+        name: "submit",
+      } %>
+    <% end %>
```

Comment:
It seems an unusual pattern to show the user the confirmation page when they have nothing to confirm - it also seems a not very friendly user pattern to hide the button (our designers often have strong opinions on this as hidden buttons is not good for knowing what you _could_ do)

What' I'd suggest is that rather than having this page deal with this error scenario we should instead check for it in the controller and redirect them back to the manual page with a flash error saying why they can't publish.

Reply:
The user shouldn't really get to this page without being able to publish (unless they've bookmarked it, or used their browser history to get back to it and the manual's state has changed in the meantime), so we thought a simple hide of the button would suffice in this case (even though it's a sub-optimal user-experience in that case), but we can implement the suggested logic in the controller.

Reply:
I'd do it in the controller - we have consistent patterns for that whereas this approach would not be an unusual pattern.

Reply:
Will do that then.

Reply:
Is a generic flash error message about not being able to publish the manual good enough, or should we specify the reason (which will mean reworking the `allow_publish?` method to not just return a boolean result)?

Reply:
OK, we've added a generic message, and addressed the other comments now.